crates/core/component/governance/src/proposal_state.rs

```diff
@@ -292,6 +292,11 @@ impl TryFrom<pb::ProposalOutcome> for Outcome<String> {
                     withdrawn,
                 }) => Outcome::Failed {
                     withdrawn: if let Some(pb::proposal_outcome::Withdrawn { reason }) = withdrawn {
+                        // Max reason length is 1kb
+                        if reason.len() > 1024 {
+                            anyhow::bail!("withdrawn reason is too long")
```

Comment:
Makes sense, i'd put the charlimit in the error message so this bubble up as a useful message